LibraryWebApp/Controllers/BookController.cs

```diff
@@ -72,40 +102,83 @@ public IActionResult Delete(int? id)
             {
                 return NotFound();
             }
+            string webRootPath = _webHostEnvironment.WebRootPath;
+            string upload = webRootPath + WC.ImagePath;
+            var oldFile = Path.Combine(upload, obj.Image);
+            if (System.IO.File.Exists(oldFile))
+            {
+                System.IO.File.Delete(oldFile);
+            }
             _db.Book.Remove(obj);
             foreach (var bookAuthor in _db.BookAuthor.Where(x => x.IdBook == id))
             {
                 
                 _db.BookAuthor.Remove(bookAuthor);
             }
+            foreach (var bookType in _db.BookType.Where(x => x.IdBook == id))
+            {
+
+                _db.BookType.Remove(bookType);
+            }
             _db.SaveChanges();
             return RedirectToAction(nameof(Index));
         }
         //GET - EDIT
         public IActionResult Edit(int? id)
         {
-            BookAuthorVM model = new BookAuthorVM
+            BookVM model = new BookVM
             {
                 Authors = _db.Author,
                 BookAuthors = _db.BookAuthor.Where(x => x.IdBook == id),
-                Book = _db.Book.Include(x => x.BookAuthor).ThenInclude(x => x.Author).Where(x => x.Id == id).FirstOrDefault()
+                Types = _db.TypeOfBook,
+                BookTypes = _db.BookType.Where(x=>x.IdBook == id),
+                Book = _db.Book.Include(x => x.BookAuthor).ThenInclude(x => x.Author).Where(x => x.Id == id).Include(x => x.BookType).ThenInclude(x => x.TypeOfBook).Where(x => x.Id == id).FirstOrDefault()
```

Comment:
you can break those functions into multiple lines so that they would not form a long snake. Add new line before '.'